shell/common/diagnostic/diagnostic_server.cc

```diff
@@ -72,16 +74,22 @@ void DiagnosticServer::Start(uint32_t port, bool ipv6) {
   EmbedderResources resources(
       &flutter::runtime::__sky_embedder_diagnostic_server_resources_[0]);
 
-  const char* source = nullptr;
-  int source_length =
-      resources.ResourceLookup(kDiagnosticServerScript, &source);
-  FXL_DCHECK(source_length != EmbedderResources::kNoSuchInstance);
-
-  Dart_Handle diagnostic_library = Dart_LoadLibrary(
-      Dart_NewStringFromCString("dart:diagnostic_server"), Dart_Null(),
-      Dart_NewStringFromUTF8(reinterpret_cast<const uint8_t*>(source),
-                             source_length),
-      0, 0);
+  Dart_Handle diagnostic_library;
+  if (running_from_kernel) {
```

Comment:
Why is it necessary to wire it this way using a parameter, is it not sufficient for this check to be
if (blink::GetKernelPlatformBinary() != nullptr) {
  ....

Reply:
Done